spv/sync.go

```diff
@@ -0,0 +1,1420 @@
+				if s.locatorGeneration > generation {
+					locators = s.currentLocators
+				}
+				if len(locators) == 0 {
+					locators, err = w.BlockLocators(ctx, nil)
+					if err != nil {
+						s.locatorMu.Unlock()
+						return err
+					}
+					s.currentLocators = locators
+					s.locatorGeneration++
+					generation = s.locatorGeneration
+				}
+				s.locatorMu.Unlock()
+				continue
+			}
+			log.Debugf("[%d] Fetched %d new header(s) ending at height %d from %v",
+				walletID, added, nodes[len(nodes)-1].Header.Height, rp)
+
+			bestChain, err := w.EvaluateBestChain(ctx, &s.sidechains)
+			if err != nil {
+				s.sidechainMu.Unlock()
+				return err
+			}
+			if len(bestChain) == 0 {
+				s.sidechainMu.Unlock()
+				continue
+			}
+
+			_, err = w.ValidateHeaderChainDifficulties(ctx, bestChain, 0)
+			if err != nil {
+				s.sidechainMu.Unlock()
+				return err
+			}
+
+			prevChain, err := w.ChainSwitch(ctx, &s.sidechains, bestChain, nil)
+			if err != nil {
+				s.sidechainMu.Unlock()
+				return err
+			}
+
+			if len(prevChain) != 0 {
+				log.Infof("[%d] Reorganize from %v to %v (total %d block(s) reorged)",
+					walletID, prevChain[len(prevChain)-1].Hash, bestChain[len(bestChain)-1].Hash, len(prevChain))
+				for _, n := range prevChain {
+					s.sidechains.AddBlockNode(n)
+				}
+			}
+			tip := bestChain[len(bestChain)-1]
+			if len(bestChain) == 1 {
+				log.Infof("[%d] Connected block %v, height %d", walletID, tip.Hash, tip.Header.Height)
+			} else {
+				log.Infof("[%d] Connected %d blocks, new tip %v, height %d, date %v",
+					walletID, len(bestChain), tip.Hash, tip.Header.Height, tip.Header.Timestamp)
+			}
+
+			s.fetchHeadersProgress(headers[len(headers)-1])
+
+			s.sidechainMu.Unlock()
+		}
+
+		// Generate new locators
+		s.locatorMu.Lock()
+		locators, err = lowestChainWallet.BlockLocators(ctx, nil)
+		if err != nil {
+			s.locatorMu.Unlock()
+			return err
+		}
+		s.currentLocators = locators
+		s.locatorGeneration++
+		s.locatorMu.Unlock()
+	}
+}
+
+func (s *Syncer) fetchMissingCFilters(ctx context.Context, rp *p2p.RemotePeer) error {
+	for walletID, w := range s.wallets {
+		s.fetchMissingCfiltersStart(walletID)
+		progress := make(chan wallet.MissingCFilterProgress, 1)
+		go w.FetchMissingCFiltersWithProgress(ctx, rp, progress)
+
+		for p := range progress {
+			if p.Err != nil {
+				return p.Err
+			}
+			s.fetchMissingCfiltersProgress(walletID, p.BlockHeightStart, p.BlockHeightEnd)
+		}
+		s.fetchMissingCfiltersFinished(walletID)
+	}
+	return nil
+}
+
+func (s *Syncer) startupSync(ctx context.Context, rp *p2p.RemotePeer) error {
+	log.Infof("Syncing %d wallets", len(s.wallets))
+
+	_, _, highestChainWallet := s.highestChainTip(ctx)
+
+	// Disconnect from the peer if their advertised block height is
+	// significantly behind the wallet's.
+	_, tipHeight := highestChainWallet.MainChainTip(ctx)
+	if rp.InitialHeight() < tipHeight-6 {
+		return errors.E("peer is not synced")
+	}
+
+	s.fetchMissingCFilters(ctx, rp)
```

Comment:
`s.fetchMissingCFilters` returns an error that should not be ignored. If the error returned is not nil, it should be returned here.
```suggestion
	if err := s.fetchMissingCFilters(ctx, rp); err != nil {
		return err
	}
```